panimg/models.py

```diff
@@ -257,6 +261,15 @@ def add_value_range_meta_data(cls, image: Image):  # noqa: B902, N805
 
         return image
 
+    @property
+    def segments(self) -> Optional[Tuple[Any, ...]]:
+        segments = np.unique(GetArrayViewFromImage(self.image))
```

Comment:
This operation should be avoided if the METAIO image type is not one of the Mask types (https://github.com/Kitware/MetaIO/blob/ffe3ce141c5a2394e40a0ecbe2a667cc0566baf5/src/metaTypes.h#L63-L94 - select the subset that are actually used as masks from there)

Reply:
The only way I could come up with to do this was the np dtype check I included. I could not find a way to extract the image type from the sitk image.

Reply:
Several ways to do it, doing the work and checking the return type is not it. The element data type is in the itk headers. The data type is a property of the numpy array.